enterprise_subsidy/apps/content_metadata/api.py

```diff
@@ -124,21 +124,52 @@ def get_geag_variant_id_for_content(self, content_identifier, content_data):
                 variant_id = additional_metadata.get('variant_id')
         return variant_id
 
+    def enroll_by_date_for_content(self, content_data, content_mode):
+        """
+        Determines the enrollment deadline for the given ``content_data``,
+        which could be either a course or course run, hence the branching
+        on content type below.
+
+        Args:
+            content_data: A dictionary of metadata pertaining to a course or course run.
+            content_mode: The (already computed) course mode for the content.
+
+        Returns:
+            A datetime string, or possibly null.
+        """
+        if content_data.get('content_type') == 'course':
+            return course_metadata.get('normalized_metadata', {}).get('enroll_by_date')
+
+        if content_mode == CourseModes.EDX_VERIFIED.value:
+            seats = content_data.get('seats', [])
+            seats_for_mode = [
+                seat for seat in content_data.get('seats', [])
```

Comment:
nit/clarification: should this be referencing the `seats` list created above on line 144?